RequestHandler.php

```diff
@@ -92,9 +92,12 @@ public function handleData($data)
      */
     public function getNextSlave()
     {
-        $slave =& $this->processManager->getNextSlave();
+        $available = SlavePool::getInstance()->getByStatus(Slave::READY);
+
+        if (count($available)) {
+            // pick random slave
+            $slave = $available[array_rand($available)];
```

Comment:
I'd not do it in a random fashion, but go through the list and pick the first available. That makes it deterministic and so easier to understand/debug.

Reply:
Good point. Will fix this.